### DISCUSSION TITLE
Condense if statement to one liner

### DISCUSSION THREAD

**Discussion by dpetka2001** [fish-shell/fish-shell#9782] [General]
I recently tried to install `docker compose` with the image from docker and it says that whenever i want to update it i should run `docker pull linuxserver/docker-compose:"${DOCKER_COMPOSE_IMAGE_TAG:-v2}"`. I looked around to see if there is something like a ternary operator for fish parameter expansion but couldn't find any info. I ended up with the following
```
docker pull linuxserver/docker-compose:(if set -q DOCKER_COMPOSE_IMAGE_TAG                                                         ─╯
       echo $DOCKER_COMPOSE_IMAGE_TAG
   else
       echo "v2"
   end)
```
and it works fine for what I want, but I'm wondering if there's a more concise way to write it. Something like `[ set -q DOCKER_COMPOSE_IMAGE_TAG ] && echo true || echo false` for example.

**Comment by faho** [comment]
My protip when it comes to the mythical one-liner is to simply not bother.

One-liners are less readable and the obsession with them leads people to write terrible code.

That being said:

```fish
docker pull linuxserver/docker-compose:(set -q DOCKER_COMPOSE_IMAGE_TAG; and echo $DOCKER_COMPOSE_IMAGE_TAG; or echo v2)
```

There, that's technically a one-liner. You can also replace the `; and` with `&&` and the `; or` with `||`. The issue with long `and`/`or` chains is the same as in bash - if that first echo fails, the second one is executed. For `echo` that's not a big deal, but for other commands it might - e.g. `foo && cat || rm` would run the `rm` if *either` the `foo` or the `cat` failed.

Here's how I would write this:

```fish
set -q DOCKER_COMPOSE_IMAGE_TAG
or set -l DOCKER_COMPOSE_IMAGE_TAG v2
docker pull linuxserver/docker-compose:$DOCKER_COMPOSE_IMAGE_TAG
```

Note that all of these fail if DOCKER_COMPOSE_IMAGE_TAG is the empty list (just `set DOCKER_COMPOSE_IMAGE_TAG`). You can use `set -q DOCKER_COMPOSE_IMAGE_TAG[1]` to see if it has at least one element (and then you probably want to use `$DOCKER_COMPOSE_IMAGE_TAG[1]` later to use only that element, in case it has multiple).

**Reply by dpetka2001** [reply]
Thank you so much for your insight.
